### ISSUE TITLE
Forced to enter (possibly) redundant information, suggesting new workflow for UI

### ISSUE THREAD

**Issue by Wulfmorn** [westnordost/StreetComplete#1017]
I have an issue with how StreetComplete gives you quests. Specifically regarding cycleways.

The issue I am raising is that StreetComplete is currently a contributor in adding "cycleway:both=no". This type of information can be defined as either implicit or unknown, but that definition depends on the commonality of yes vs. no. If there is a 50-50 split between yes and no the implicit answer is "unknown", if it is 90-10 the implicit is no.

In the case of bike lanes I would suggest that the ratio of roads WITH bike lanes is in vast minority and that all roads should be considered to NOT have bike lanes unless stated otherwise. 

Adding "no"-tags to things is bad practice for most things, like "oneway=no", "boat=no" or on the reverse side "motor_vehicle=yes" on major roads.

----

Now to my issue with StreetComplete as an app. The way the app works right now is to show you the quests you have enabled in sequence. You have to do one quest in order to get the next one. This means that if you want to set something like road surface or lit you have to do the bike lanes first (I dont know the specific order but this is an example). That means you either have to go through the "I dont know, leave not, yes/no" ordeal or in the case of bike lanes say "no".

I would suggest that each item WITH quests to them should have an icon "this has quests" and let the user click that and pick which quest to do. That way house numbers dont block opening hours, road speeds dont block road surface - etc. It would simply be more handy.

That would make it easier to map while moving. Even while walking it can take longer to hide quests and tweak the app than it takes to walk the street, and quite impossible if you are in a car (or on a train - which I have tried).

----

tl;dr - let the user pick which quest to do on each item rather than have them in sequence, this will avoid unnecessary tagging and notes and allow a faster UI while mapping while moving.

**Comment by matkoniecz** [comment]
https://wiki.openstreetmap.org/wiki/StreetComplete/FAQ#Why_does_StreetComplete_often_tag_the_absence_of_features.3F

**Comment by matkoniecz** [comment]
> tl;dr - let the user pick which quest to do on each item rather than have them in sequence, this will avoid unnecessary tagging and notes and allow a faster UI while mapping while moving.

Are you aware that it is possible to change priority of quests in settings and disable unwanted ones?

**Comment by Wulfmorn** [comment]
I didnt know about sorting them but I did know about disable. But thats what I mean with "tweak the app". Sometimes I come across a road that HAS a cycleway - and then I have to enable that quest in order to see if is a current quest or not - AND make sure that that quest isnt hidden by some other quest.

Just tell me that "this road has possible quests" and let me pick which one i want to do.

**Comment by westnordost** [comment]
This is a reasonable feature request, but iirc correctly it has been requested before. The reason why it cannot be done right now is because it would requires implementing changes closely related to the map rendering. However, any features regarding the map rendering are deferred indefinitely as long as the future of the map rendering library tangram-es is not clear yet.

**Comment by westnordost** [comment]
Regarding tagging "no", @matkoniecz hit the nail with posting the FAQ entry.

**Comment by Wulfmorn** [comment]
I see the FAQ makes specific cases on this topic, and I would say that one can make arguments for both sides about the bike lanes.

**Comment by Wulfmorn** [comment]
> As for the UI, I'd already suggested "it would be nice to let the user cycle through all quests available", so offer a "<" and a ">" button, so one can easily choose one.

Yes. I imagined just one icon on the map that opens a menu-ish list or dropdown (one way or another) to pick the quest you wish to resolve.

**Comment by ENT8R** [comment]
If you have another suggestion which was not yet suggested in #124 you should post it there because this issue here has been already closed.
